app/controllers/concerns/encryptables.rb

```diff
@@ -56,6 +56,49 @@ def credential_id
     params[:credential_id]
   end
 
+  ### Sharing ###
+
+  def encryptable_transfering?
+    receiver_id.present?
+  end
+
+  def transfer_encryptable
+    sender_id = current_user.id
+
+    shared_encryptable = EncryptableTransfer.new.transfer(encryptable, receiver_id, sender_id)
+
+    instance_variable_set(:"@#{ivar_name}", shared_encryptable)
+  end
+
+  def transfered_encryptable?(entry)
+    entry.encrypted_transfer_password.present? && entry.sender_id.present?
+  end
+
+  def receiver_id
+    params.dig('data', 'attributes', 'receiver_id')
+  end
+
+  def decrypt_shared_encryptable(entry, private_key)
+    plaintext_transfer_password = Crypto::Rsa.decrypt(entry.transfer_password, private_key)
+    entry.decrypt(plaintext_transfer_password)
+
+    recrypt_with_personal_team_password(entry)
+    remove_shared_attributes(entry)
```

Comment:
remove transfer attributes when doing a recrypt inside encryptable model. 
this makes sure the encryptable entry cannot be in an invalid state (recrypted, but still sender_id and transfer_password set)